---
title: "EHP Shen Perinatal Exposure-Child Microbiome_pathway analysis"
author: "Yike Shen"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
---
## Readme:
#You can not run this script without metadata, this aims to provide the workflow of our paper and offer pseudo code example for analysis. 
# Request for data must go through approval with GESTE cohort, after approval I can share the original scripts with participants name and input processed sequence data. 
#All participant names are de-identified as "PARTICIPANT_ID"; All name of the inputsheets are redacted as "INPUTDATA.csv"
## End

#This is pathway analysis for shotgun metagenome data FOR Perinatal Exposure-Child Microbiome

#metadata preprocessing
```{r}
GESTERAW <- read.csv("$path/GESTERAWMothers.csv")
GESTESEX <- read.csv("$path/GESTEY6RAW.csv") %>% select(record_id,sexe_enfan)

IDLIST <- GESTERAW[,1] %>% as.data.frame()
Deliverycon <- GESTERAW[,c(7:11)] %>% as.data.frame()
ChildMetalRaw <- c(IDLIST,Deliverycon) %>% as.data.frame()

ChildMetalprocessing<- apply(ChildMetalRaw, 2, function(x) gsub("^$|^ $", NA, x)) %>% as.data.frame()

ChildMetalprocess <- ChildMetalprocessing %>% as.tibble() %>%
  dplyr::rename(Cd_D=cd_acc) %>% 
  dplyr::rename(Mn_D=mn_acc) %>% 
  dplyr::rename(Hg_D=hgt_acc) %>% 
  dplyr::rename(Pb_D=pb_acc) %>% 
  dplyr::rename(Se_D=se_acc) %>% 
  dplyr::rename(SampleID=".")

#more covariates from the other datasheet
Covariates <- read.csv("$path/Mothercovariates.csv")
mothercovariates <- Covariates[,c(15:197,198)]

GESTERAWwcovaraites <- cbind(ChildMetalprocess,mothercovariates)
GESTERAWwcovaraites <- GESTERAWwcovaraites[-1,]
GESTERAWwcovaraites <- cbind(GESTERAWwcovaraites,GESTESEX)

#############Diagnoses################
#pregnancy related diagnoses
diagnosesprg <- GESTERAWwcovaraites[,8:9]
diagnosesprg[] <- lapply(diagnosesprg, function(x) as.numeric(as.character(x)))
diagnosesprglist <- rowSums(diagnosesprg) %>% as.data.frame()
colnames(diagnosesprglist) <- "DiagnoseIndex_prgnancy"

#preexisting diagnoses
diagnoses <- GESTERAWwcovaraites[,10:16]
diagnoses[] <- lapply(diagnoses, function(x) as.numeric(as.character(x)))
diagnoseslist <- rowSums(diagnoses) %>% as.data.frame()
colnames(diagnoseslist) <- "DiagnoseIndex_preexisting"

diagnoseTWO <- cbind(diagnosesprglist,diagnoseslist)
diagnoseTWO[] <- lapply(diagnoseTWO, function(x) as.factor(as.numeric(x)))
table1::table1(~., data = diagnoseTWO)

####################Exposure###############
riskfactors <- GESTERAWwcovaraites[,46:157]
riskfactors[] <- lapply(riskfactors, function(x) as.numeric(as.character(x)))

num_column2<-112
num_row2<-86

sum_df<-data.frame(matrix(0,nrow=86,ncol=28))
for (i in 1:num_row2){
  for (j in 1:((num_column2)/4)){
    Process_vector<-riskfactors[i, c(j*4-3,j*4-2,j*4-1,j*4)]
      sumdiagnose <- rowSums(Process_vector)
    sum_df[i,j] = sumdiagnose
  }
}
   
Colnamesexposure <- cbind("dust","smoke","emanations","odors","combustion","leather_dust","printing_inks", "metal_dust", "tars/bitumen", "oils/grease", "fuel/heating_oil","non-flammable_sheaths","fertilizers", "pesticides","paints", "photographic_goods", "colles","solvent/degreasing_fluids", "radiation", "industrial_phamaceuticals", "industrial_food_products", "antioxidnts","dyes", "heat", "cold", "temp","humidity", "noise")

colnames(sum_df) <- Colnamesexposure
sum_df[] <- lapply(sum_df, function(x) as.factor(as.numeric(x)))

table1::table1(~., data = sum_df)

importantcov <- GESTERAWwcovaraites %>% select(SampleID,acc_cs,bmipostpregn,revenu_delivery,ga_wks_bb,sexe_enfan) %>% 
  dplyr::rename(Delivery_mode=acc_cs) %>% 
  dplyr::rename(BMI_delivery=bmipostpregn) %>% 
  dplyr::rename(Family_Income=revenu_delivery) %>% 
  dplyr::rename(Gestational_age=ga_wks_bb) %>% 
  dplyr::rename(Sex=sexe_enfan)

importantcov <- cbind(importantcov,diagnoseTWO)

#Missing income, impute from year6 income, for those without year6 income, impute with median 65000
importantcov[11,4] <- NUMBER #participant PARTICIPANT_ID
importantcov[16,4] <- NUMBER #participant PARTICIPANT_ID
importantcov[32,4] <- NUMBER#participant PARTICIPANT_ID, NA fron Y6 impute with median
importantcov[35,4] <- NUMBER#participant PARTICIPANT_ID
importantcov[62,4] <- NUMBER #participant PARTICIPANT_ID
importantcov[82,4] <- NUMBER#participant PARTICIPANT_ID, NA from Y6 impute with median

importantcov$Sex[importantcov$Sex == "1"] <- 'Female'
importantcov$Sex[importantcov$Sex == "2"] <- 'Male'

##ATTENTION!!!####CHANGE THE IMPUTATION AFTER ANNIE GETS BACK ON ME#######
#importantcov[71,6] <- 'Male'
###############
importantcov$DiagnoseIndex_prgnancy <- as.character(importantcov$DiagnoseIndex_prgnancy)
importantcov$DiagnoseIndex_prgnancy[importantcov$DiagnoseIndex_prgnancy == "1"] <- 'Yes'
importantcov$DiagnoseIndex_prgnancy[importantcov$DiagnoseIndex_prgnancy == "2"] <- 'Yes'
importantcov$DiagnoseIndex_prgnancy[importantcov$DiagnoseIndex_prgnancy == "0"] <- 'No'

importantcov$DiagnoseIndex_preexisting <- as.character(importantcov$DiagnoseIndex_preexisting)
importantcov$DiagnoseIndex_preexisting[importantcov$DiagnoseIndex_preexisting == "1"] <- 'Yes'
importantcov$DiagnoseIndex_preexisting[importantcov$DiagnoseIndex_preexisting == "2"] <- 'Yes'
importantcov$DiagnoseIndex_preexisting[importantcov$DiagnoseIndex_preexisting == "3"] <- 'Yes'
importantcov$DiagnoseIndex_preexisting[importantcov$DiagnoseIndex_preexisting == "0"] <- 'No'

Socio <- GESTERAWwcovaraites %>% select(SampleID,Cd_D,Mn_D,Hg_D,Pb_D, Se_D,age_deliverymoth)
covariatesmotherdelivery <- cbind(Socio,importantcov)

Socioprocessing <- covariatesmotherdelivery[complete.cases(covariatesmotherdelivery[ , 2:6]),]

#remove the wrong data
Socioprocessing <- Socioprocessing[!(Socioprocessing$SampleID %in% "PARTICIPANT_ID"), ]

Socioprocessing <- Socioprocessing[,-c(2:6)]
Socioprocessing <- Socioprocessing %>% as.tibble() %>% dplyr::rename(Age=age_deliverymoth)
Socioprocessing <- Socioprocessing[,-3]

row.names(Socioprocessing) <- Socioprocessing$SampleID
Socioprocessing[,c(1:2,4:6)] <- lapply(Socioprocessing[,c(1:2,4:6)], function(x) as.numeric(as.character(x)))
Socioprocessing$Delivery_mode[Socioprocessing$Delivery_mode == "1"] <- 'Vaginal'
Socioprocessing$Delivery_mode[Socioprocessing$Delivery_mode == "2"] <- 'C-Section'
row.names(Socioprocessing) <- Socioprocessing$SampleID


map <- read_excel("$path/GESTE_OTU.xlsx", sheet = "ENV_MotherD")
SAMPLEIDROWNAMES <- map$SampleID
map <- map %>% as.matrix()
row.names(map) <- SAMPLEIDROWNAMES
map <- map[,c(-1,-7)]
ROANAMESLISTSAMPLESID <- row.names(map)
map <- map %>% as.data.frame()
map <- map[!(row.names(map) %in% "2336"), ]

dataframewcovariates <- cbind(map,Socioprocessing)
dataframewcovariates <- dataframewcovariates[,-6]
dataframewcovariates[,1:5] <- lapply(dataframewcovariates[,1:5], function(x) as.numeric(x))
```

#preprocessing, pathway
```{r}
FUNC <- lapply(list.files(pattern = "*humann_pathway_relab.tsv"), function(i){read_tsv(i)}) 

IDS <- lapply(FUNC,function(x) substring(names(x),1,regexpr("_dehost_Abundance",names(x))-1))
IDS <- unlist(IDS)[-1]

FUNC <- lapply(FUNC, function(x)
  separate(data = x, col = "# Pathway", into = c("Pathway", "Taxon"), sep = "\\|"))
FUNC <- lapply(FUNC, subset, is.na(Taxon))

FUNC <- FUNC %>%
  reduce(full_join, by = "Pathway")
FUNC <- FUNC[,which(!grepl("Taxon", colnames(FUNC)))] 

colnames(FUNC)[2:ncol(FUNC)] <- IDS
Pathways <- FUNC$Pathway
names(Pathways) <- ifelse(grepl(":", Pathways), substring(Pathways,1,regexpr(":",Pathways)-1), Pathways)
FUNC <- FUNC[,-1]
FUNC <- as.data.frame(t(FUNC))
FUNC[is.na(FUNC)] <- 0
colnames(FUNC) <- names(Pathways)
```

#preprocessing, merge metadata and functional pathway
```{r}
MATCHING <- rownames(dataframewcovariates) %>% as.data.frame()
colnames(MATCHING) <- "ID"
MATCHING <- cbind(MATCHING,dataframewcovariates)

FUNCMATCHING <- rownames(FUNC) %>% as.data.frame()
colnames(FUNCMATCHING) <- "ID"
FUNCXX <- cbind(FUNCMATCHING,FUNC)

#Matching missing samples
FUNCXX <- match_df(FUNCXX, MATCHING, on ="ID")
MATCHING <- match_df(MATCHING,FUNCXX, on ="ID")

FUNCXX <- FUNCXX[,colSums(FUNCXX == 0) <= 62]

input_metadata_phy <- MATCHING %>% as.data.frame()
input_data_phy <- FUNCXX %>% as.data.frame()
rownames(input_metadata_phy) <- rownames(input_data_phy)
input_data_phy <- input_data_phy[,-1]

#remove outliers
input_metadata_phy_Pb <- input_metadata_phy[!(row.names(input_metadata_phy) %in% "PARTICIPANT_ID"), ]
input_data_phy_Pb <- input_data_phy[!(row.names(input_data_phy) %in% "PARTICIPANT_ID"), ]

input_metadata_phy_Cd <- input_metadata_phy[!(row.names(input_metadata_phy) %in% "PARTICIPANT_ID"), ]
input_data_phy_Cd <- input_data_phy[!(row.names(input_data_phy) %in% "PARTICIPANT_ID"), ]
input_metadata_phy_Cd <- input_metadata_phy_Cd[!(row.names(input_metadata_phy_Cd) %in% "PARTICIPANT_ID"), ]
input_data_phy_Cd <- input_data_phy_Cd[!(row.names(input_data_phy_Cd) %in% "PARTICIPANT_ID"), ]

```

#pathway association
```{r}
#361 pathways

#Cd
fit_data1 = Maaslin2(
    input_data = input_data_phy_Cd,
    input_metadata = input_metadata_phy_Cd,
    output = "masslin_pathway_Cd_mother",
    fixed_effects = c("Cd_D_c","Delivery_mode","BMI_delivery","DiagnoseIndex_prgnancy","DiagnoseIndex_preexisting","Sex","Family_Income"))
FDRq_Cd <- read.table(file="./masslin_pathway_Cd_mother/all_results.tsv",sep = '\t', header = TRUE)
FDRq_Cd <- FDRq_Cd[which(FDRq_Cd$metadata=='Cd_D_c', ),]
correctedq_Cd <- p.adjust(FDRq_Cd$pval, method = "fdr")
FDRq_Cd <- cbind(FDRq_Cd,correctedq_Cd)

#Mn
fit_data2 = Maaslin2(
    input_data = input_data_phy,
    input_metadata = input_metadata_phy,
    output = "masslin_pathway_Mn_M",
    fixed_effects = c("Mn_D_c","Delivery_mode","BMI_delivery","DiagnoseIndex_prgnancy","DiagnoseIndex_preexisting","Sex","Family_Income"))
FDRq_Mn <- read.table(file="./masslin_pathway_Mn_M/all_results.tsv",sep = '\t', header = TRUE)
FDRq_Mn <- FDRq_Mn[which(FDRq_Mn$metadata=='Mn_D_c', ),]
correctedq_Mn <- p.adjust(FDRq_Mn$pval, method = "fdr")
FDRq_Mn <- cbind(FDRq_Mn,correctedq_Mn)

#Hg
fit_data3 = Maaslin2(
    input_data = input_data_phy,
    input_metadata = input_metadata_phy,
    output = "masslin_pathway_Hg_M",
    fixed_effects = c("Hg_D_c","Delivery_mode","BMI_delivery","DiagnoseIndex_prgnancy","DiagnoseIndex_preexisting","Sex","Family_Income"))
FDRq_Hg <- read.table(file="./masslin_pathway_Hg_M/all_results.tsv",sep = '\t', header = TRUE)
FDRq_Hg <- FDRq_Hg[which(FDRq_Hg$metadata=='Hg_D_c', ),]
correctedq_Hg <- p.adjust(FDRq_Hg$pval, method = "fdr")
FDRq_Hg <- cbind(FDRq_Hg,correctedq_Hg)

#Pb
#P161.PWY: P161-PWY: acetylene degradation
#PWY4LZ.257: PWY4LZ-257: superpathway of fermentation (Chlamydomonas reinhardtii)
fit_data4 = Maaslin2(
    input_data = input_data_phy_Pb,
    input_metadata = input_metadata_phy_Pb,
    output = "masslin_pathway_Pb_M",
    fixed_effects = c("Pb_D_c","Delivery_mode","BMI_delivery","DiagnoseIndex_prgnancy","DiagnoseIndex_preexisting","Sex","Family_Income"))
FDRq_Pb <- read.table(file="./masslin_pathway_Pb_M/all_results.tsv",sep = '\t', header = TRUE)
FDRq_Pb <- FDRq_Pb[which(FDRq_Pb$metadata=='Pb_D_c', ),]
correctedq_Pb <- p.adjust(FDRq_Pb$pval, method = "fdr")
FDRq_Pb <- cbind(FDRq_Pb,correctedq_Pb)

#Se
fit_data5 = Maaslin2(
    input_data = input_data_phy,
    input_metadata = input_metadata_phy,
    output = "masslin_pathway_Se_M",
    fixed_effects = c("Se_D_c","Delivery_mode","BMI_delivery","DiagnoseIndex_prgnancy","DiagnoseIndex_preexisting","Sex","Family_Income"))
FDRq_Se <- read.table(file="./masslin_pathway_Se_M/all_results.tsv",sep = '\t', header = TRUE)
FDRq_Se <- FDRq_Se[which(FDRq_Se$metadata=='Se_D_c', ),]
correctedq_Se <- p.adjust(FDRq_Se$pval, method = "fdr")
FDRq_Se <- cbind(FDRq_Se,correctedq_Se)

FDRq_Cd <- FDRq_Cd %>% dplyr::rename(FDR_qval=correctedq_Cd)
FDRq_Mn <- FDRq_Mn %>% dplyr::rename(FDR_qval=correctedq_Mn)
FDRq_Hg <- FDRq_Hg %>% dplyr::rename(FDR_qval=correctedq_Hg)
FDRq_Pb <- FDRq_Pb %>% dplyr::rename(FDR_qval=correctedq_Pb)
FDRq_Se <- FDRq_Se %>% dplyr::rename(FDR_qval=correctedq_Se)


Allassociationspathway <- rbind(FDRq_Cd,FDRq_Mn,FDRq_Hg,FDRq_Pb,FDRq_Se)
Allassociationspathway <- Allassociationspathway %>% select(metadata, feature, coef,stderr,pval,FDR_qval)
Allassociationspathway[,3:6] <- Allassociationspathway[,3:6] %>% round(3)

write.csv(Allassociationspathway,"associationspathway_mc.csv")
```


#plot association, evaluation
```{r}
rownames(FDRq_Pb) <- c()
FDRq_mc <- FDRq_Pb[c(1,2,3),]
FDRq_mc$feature

#PWY-6895: superpathway of thiamin diphosphate biosynthesis II
    #contributing bacteria: g__Bacteroides.s__Bacteroides_xylanisolvens; 
    #g__Escherichia.s__Escherichia_coli;
    #g__Klebsiella.s__Klebsiella_pneumoniae
#PWY-6147: 6-hydroxymethyl-dihydropterin diphosphate biosynthesis I
#PWY-7539: 6-hydroxymethyl-dihydropterin diphosphate biosynthesis III (Chlamydia) 

Pathwayfullname <- c("superpathway of thiamin diphosphate biosynthesis II","6-hydroxymethyl-dihydropterin diphosphate biosynthesis I","6-hydroxymethyl-dihydropterin diphosphate biosynthesis III (Chlamydia)")

Stats_mc <- cbind(FDRq_mc,Pathwayfullname)
clean_mc_stat <- Stats_mc %>% select(metadata,feature,Pathwayfullname,value,coef,pval,correctedq_Pb)

PWY.6895 <- input_data_phy %>% select("PWY-6895")
PWY.6147 <- input_data_phy %>% select("PWY-6147")
PWY.7539 <- input_data_phy %>% select("PWY-7539")
Pb_maasline <- input_metadata_phy %>% select("Pb_D_c")
pathwaydf <- cbind(PWY.6895,PWY.6147,PWY.7539,Pb_maasline) %>% as.data.frame()
pathwaydf <- cbind(pathwaydf,input_metadata_phy[,7:14])
colnames(pathwaydf) <- gsub("-", "_", colnames(pathwaydf))

#PARTICIPANT_ID lead concentration too high
pathwayplot1 <- ggplot(pathwaydf,aes(x=Pb_D_c, y=PWY_6895)) +
  geom_point() + geom_rug()+
  geom_smooth(method=lm, se=FALSE)+theme_bw()+
  ggtitle("Association of superpathway of thiamin diphosphate biosynthesis II ~ children metal exposure, adjusting for delivery mode, bmi, sex, family income, diagnose index_pregnancy, diagnose index preexisting")

summary(piCd <- lm( PWY_6895~ Pb_D_c+Delivery_mode+BMI_delivery+Family_Income+Sex+DiagnoseIndex_prgnancy+DiagnoseIndex_preexisting, data=pathwaydf))
par(mfrow=c(2,2))
plot(piCd,which=1:4,main="regression diagnose_Pb_D_c~superpathway of thiamin diphosphate biosynthesis II")

pathwayplot2 <- ggplot(pathwaydf,aes(x=Pb_D_c, y=PWY_6147)) +
  geom_point() + geom_rug()+
  geom_smooth(method=lm, se=FALSE)+theme_bw()+
  ggtitle("Association of superpathway of fermentation (Chlamydomonas reinhardtii) ~ children metal exposure, adjusting for breastfeeding, family income, gestational age, sex")

summary(piCd <- lm( PWY_6147~ Pb_D_c+Delivery_mode+BMI_delivery+Family_Income+Sex+DiagnoseIndex_prgnancy+DiagnoseIndex_preexisting, data=pathwaydf))
par(mfrow=c(2,2))
plot(piCd,which=1:4,main="regression diagnose_Pb_D_c~superpathway of thiamin diphosphate biosynthesis II")

pathwayplot3 <- ggplot(pathwaydf,aes(x=Pb_D_c, y=PWY_7539)) +
  geom_point() + geom_rug()+
  geom_smooth(method=lm, se=FALSE)+theme_bw()+
  ggtitle("Association of superpathway of fermentation (Chlamydomonas reinhardtii) ~ children metal exposure, adjusting for breastfeeding, family income, gestational age, sex")

summary(piCd <- lm( PWY_7539~ Pb_D_c+Delivery_mode+BMI_delivery+Family_Income+Sex+DiagnoseIndex_prgnancy+DiagnoseIndex_preexisting, data=pathwaydf))
par(mfrow=c(2,2))
plot(piCd,which=1:4,main="regression diagnose_Pb_D_c~superpathway of thiamin diphosphate biosynthesis II")

write.csv(clean_mc_stat,file="pathway_mother_children_pair.csv")
```
